src/stores/AppStore.ts

```diff
@@ -873,6 +848,97 @@ export class AppStore {
         });
     };
 
+    // Online Catalog Query
```

Comment:
you can move some of this to the newly created service class. 

Can you not just use `await` to simplify this?